src/prefect/utilities/asyncutils.py

```diff
@@ -245,7 +261,7 @@ def in_async_main_thread() -> bool:
         return not in_async_worker_thread()
 
 
-def sync_compatible(async_fn: T) -> T:
+def sync_compatible(async_fn: T, force_sync: bool = False) -> T:
```

Comment:
Where do you expect `force_sync` to be used? It looks like it isn't used in this PR.

Reply:
@desertaxle sorry didn't see this before merging -- I needed it for the CLI (which is all `sync_compatible` but is never awaited), and calling `run_sync` directly doesn't work as well because it creates nested `run_sync` calls.  (one way to think of `sync_compatible` in the new world is that it's an efficient wrapper to `run_sync`) so this approach was easier.  Very open to pushback on that, especially when I wire it up to the CLI